perses/rjmc/geometry.py

```diff
@@ -163,30 +164,31 @@ def logp_reverse(self, top_proposal, new_coordinates, old_coordinates):
             The coordinates of the system after the proposal
         old_coordiantes : [n, 3] np.ndarray
             The coordinates of the system before the proposal
+        beta : float
+            The inverse temperature
 
         Returns
         -------
         logp : float
             The log probability of the proposal for the given transformation
         """
-        beta = top_proposal.beta
         logp = 0.0
-        top_proposal = topology_proposal.SmallMoleculeTopologyProposal()
-        structure = parmed.openmm.load_topology(top_proposal.old_topology, top_proposal.old_system)
-        atoms_with_positions = [structure.atoms[atom_idx] for atom_idx in range(top_proposal.n_atoms_old) if atom_idx not in top_proposal.unique_old_atoms]
-        new_atoms = [structure.atoms[idx] for idx in top_proposal.unique_old_atoms]
-        #we'll need to copy the current positions of the core to the old system
-        #In the case of C-A --> C/-A -> C/-B ---> C-B these are the same
-        reverse_proposal_coordinates = units.Quantity(np.zeros([top_proposal.n_atoms_new, 3]), unit=units.nanometers)
-        for atom in atoms_with_positions:
-            new_index = top_proposal.old_to_new_atom_map[atom.idx]
-            reverse_proposal_coordinates[atom.idx] = new_coordinates[new_index]
+        old_structure = parmed.openmm.load_topology(top_proposal.old_topology, top_proposal.old_system)
+        old_atoms_with_positions = [old_structure.atoms[atom_idx] for atom_idx in top_proposal.old_to_new_atom_map.keys()]
+        old_unique_atoms = [old_structure.atoms[idx] for idx in top_proposal.unique_old_atoms]
+
+        #copy common atomic positions
+        for atom in old_structure.atoms:
+            if atom.idx in old_atoms_with_positions:
+                corresponding_new_index = top_proposal.old_to_new_atom_map[atom.idx]
+                old_coordinates[atom.idx] = new_coordinates[corresponding_new_index]
```

Comment:
This looks fishy to me.  What are you trying to do here---shouldn't you already have the old coordinates?

The goal of computing the reverse proposal probability is just to compute the probability of placing the old atoms that are deleted back where they were originally.